src/test/java/io/jenkins/pluginhealth/scoring/service/PluginServiceIT.java

```diff
@@ -0,0 +1,51 @@
+package io.jenkins.pluginhealth.scoring.service;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.jenkins.pluginhealth.scoring.AbstractDBContainerTest;
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.repository.PluginRepository;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.stereotype.Service;
+
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+@DataJpaTest(includeFilters = @ComponentScan.Filter(Service.class))
+public class PluginServiceIT extends AbstractDBContainerTest {
+    @Autowired
+    private PluginRepository pluginRepository;
+    @Autowired
+    private PluginService pluginService;
+
+    @Test
+    public void shouldNotDuplicatePlugin() {
+        Plugin plugin = new Plugin("myPlugin", "https://github.com/jenkinsci/my-plugin", null);
+
+        pluginService.saveOrUpdate(plugin);
+        pluginService.saveOrUpdate(plugin);
+
+        assertThat(pluginRepository.findAll())
+            .hasSize(1)
+            .contains(plugin);
+    }
```

Comment:
I think we can get rid of this test because it is not properly validating the code.